### PR TITLE
Replace `Tip` by `WithOrigin`

### PR DESCRIPTION
**Pull request by edsko**

### CODE REVIEW

File: ouroboros-consensus/src/Ouroboros/Consensus/Storage/Common.hs [hunk 1]
```diff
@@ -32,84 +31,53 @@ module Ouroboros.Consensus.Storage.Common (
...
-      0 -> return TipGen
-      1 -> Tip <$> decodeR
-      _ -> fail "decodeTip: invalid tag"
+encodeTip :: (r            -> Encoding)
+          -> (WithOrigin r -> Encoding)
+encodeTip = encodeWithOrigin
+
+decodeTip :: (forall s. Decoder s             r)
+          -> (forall s. Decoder s (WithOrigin r))
+decodeTip = decodeWithOrigin
```
Comment (mrBliss) [inline ouroboros-consensus/src/Ouroboros/Consensus/Storage/Common.hs]: Those should be removed

Comment (edsko) [reply ouroboros-consensus/src/Ouroboros/Consensus/Storage/Common.hs]: Will do so in subsequent PR, which will close #1155.

**Pull request merged**